File: 2405.1/jf_syn/pmd/AvoidCatchingGenericExceptionChecker/AvoidCatchingGenericException.java
package src.test.java.pmd;

public class AvoidCatchingGenericException {
    public void downCastPrimitiveType() {
        try {
            System.out.println(" ");
        } catch(Exception e) { // @violation
            e.printStackTrace();
        } catch(RuntimeException e) { // @violation
            e.printStackTrace();
        } catch(NullPointerException e) { // @violation
            e.printStackTrace();
        }
    }
}

File: 2405.1/crulechk/local/A/AVOID_CATCHING_GENERIC_EXCEPTION/AVOID_CATCHING_GENERIC_EXCEPTION.cpp
//@checker AVOID_CATCHING_GENERIC_EXCEPTION
class exception {
	
};

void foo(char c)
{
  try 
  {
  
  }
  catch (exception e)	//@violation AVOID_CATCHING_GENERIC_EXCEPTION
  {
  
  }
}

int main() {
	return 0;
}